rpcserver.go

```diff
@@ -1946,6 +1947,93 @@ func createRPCCloseUpdate(update interface{}) (
+	// return the error back to the caller.
+	dbChan, err := db.FetchChannel(*chanPoint)
+	switch {
+	// If the channel wasn't found, then it's possible that it was already
+	// abandoned from the database.
+	case err == channeldb.ErrChannelNotFound:
+		_, closedErr := db.FetchClosedChannel(chanPoint)
+		if closedErr != nil {
+			return closedErr
+		}
+
+		// If the channel was already closed, then we don't return an
+		// error as we'd like fro this step to be repeatable.
+		return nil
+	case err != nil:
+		return err
+	}
+
+	// Now that we've found the channel, we'll populate a close summary for
+	// the channel, so we can store as much information for this abounded
+	// channel as possible. We also ensure that we set Pending to false, to
+	// indicate that this channel has been "fully" closed.
+	_, bestHeight, err := chainIO.GetBestBlock()
+	if err != nil {
+		return err
+	}
+	summary := &channeldb.ChannelCloseSummary{
+		CloseType:               channeldb.Abandoned,
+		ChanPoint:               *chanPoint,
+		ChainHash:               dbChan.ChainHash,
+		CloseHeight:             uint32(bestHeight),
+		RemotePub:               dbChan.IdentityPub,
+		Capacity:                dbChan.Capacity,
+		SettledBalance:          dbChan.LocalCommitment.LocalBalance.ToSatoshis(),
+		ShortChanID:             dbChan.ShortChanID(),
+		RemoteCurrentRevocation: dbChan.RemoteCurrentRevocation,
+		RemoteNextRevocation:    dbChan.RemoteNextRevocation,
+		LocalChanConfig:         dbChan.LocalChanCfg,
+	}
+
+	// Finally, we'll close the channel in the DB, and return back to the
+	// caller.
+	return dbChan.CloseChannel(summary)
+}
+
+// abandonChanFromGraph attempts to remove a channel from the channel graph. If
+// we can't find the chanID in the graph, then we assume it has already been
+// removed, and will return a nop.
+func abandonChanFromGraph(chanGraph *channeldb.ChannelGraph,
+	chanPoint *wire.OutPoint) error {
+
+	// First, we'll obtain the channel ID. If we can't locate this, then
+	// it's the case that the channel may have already been removed from
+	// the graph, so we'll return a nil error.
+	chanID, err := chanGraph.ChannelID(chanPoint)
+	switch {
+	case err == channeldb.ErrEdgeNotFound:
+		return nil
+	case err != nil:
+		return err
+	}
+
+	// If the channel ID is still in the graph, then that means the channel
+	// is still open, so we'll now move to purge it from the graph.
+	return chanGraph.DeleteChannelEdges(chanID)
+}
+
+// abandonChanFromArbitrator attempts to remove all active contractcourt state
+// concerning the passed channel point. This is the last abandon channel method
+// to be called as it will fail if the channel state has already been removed.
+// This serves to notify the user if they channel has already been _fully_
+// abandoned.
+func abandonChanFromArbitrator(chainArb *contractcourt.ChainArbitrator,
+	chanPoint *wire.OutPoint) error {
+
+	return chainArb.ResolveContract(*chanPoint)
+}
```

Comment:
Does this function add anything?

Reply:
Agree with @joostjager here, as this is a one line function, unless called multiple times, you might consider inlinging it.

Reply:
When I started it, I expected to have to do more in the body, but turns out `ResolveContract` did most of the heavy lifting for us. Ultimately, I still like the function though as its `godoc` gives information to the reader which would otherwise be more scrunched in the main `AbandonChannel` method.